crates/vm-superio/src/serial.rs

```diff
@@ -304,7 +304,7 @@ impl<T: Trigger, W: Write> Serial<T, NoEvents, W> {
     ///
     /// You can see an example of how to use this function in the
     /// [`Example` section from `Serial`](struct.Serial.html#example).
-    pub fn new(trigger: T, out: W) -> Serial<T, NoEvents, W> {
+    pub fn new(trigger: T, out: W) -> Result<Serial<T, NoEvents, W>, Error<T::E>> {
```

Comment:
I think this should use unwrap and comment why the unwrap is safe so we don't need to change the existing API.